src/sdk/pynni/nni/compression/tensorflow/compressor.py

```diff
@@ -228,80 +189,61 @@ def call(self, *inputs):
 #        self._pruner.update_mask()
 
 
-def _detect_layers_to_compress(model, config_list):
-    # Returns list of LayerInfo.
-    located_layers = _locate_layers(model)
-    ret = []
-    for layer in model.layers:
-        config = _select_config(LayerInfo(layer), config_list)
-        if config is not None:
-            if id(layer) not in located_layers:
-                _logger.error('Failed to locate layer %s in model. The layer will not be compressed. '
-                              'This is a bug in NNI, feel free to fire an issue.', layer.name)
-                continue
-            layer_info = located_layers[id(layer)]
-            layer_info.config = config
-            ret.append(layer_info)
-    return ret
-
-def _locate_layers(model, cur_path=[]):
-    # Find out how to access layers from model object.
-    # Returns dict of (layer's object ID, LayerInfo).
-    # This function is required because TF framework does not track layer's attribute name,
-    # and to my knowledge `Layer.name` is only useful for read-only access.
-    # `cur_path`s format is documented in `LayerInfo.path`.
-    # TODO: it can only find layers in `Model` and `list` for now.
-    assert isinstance(model, tf.keras.Model)
-    if isinstance(model, tf.keras.Sequential):
-        _logger.warning('`Sequential` model is not supported yet, ignored.')
-    ret = {}
-    for key, value in model.__dict__.items():
-        if isinstance(value, tf.keras.Model):
-            ret.update(_locate_layers(value, cur_path + [key]))
-        elif isinstance(value, tf.keras.layers.Layer):
-            ret[id(value)] = LayerInfo(value, cur_path + [key])
+def _instrument(layer, config_list, LayerWrapperClass, compressor):
```

Comment:
Is it possible to move `_instrument` and other functions into `Compressor` class?

Reply:
Yes, of course. I just personally don't like heavy methods which are used in (and can only be used in) constructor. Because I think "partially constructed" objects can be confusing.